### DISCUSSION TITLE
How can I use gltf multiple animations in combination？ and now I find that it is overwritten by the last animation。

### DISCUSSION THREAD

**Discussion by caorm** [google/filament#5650] [General]
now I find that it is overwritten by the last animation，for example:

size_t animationCount = animator->getAnimationCount();
for (size_t i = 0; i < animationCount; ++i) {
    animator->applyAnimation(i, static_cast<float>(elapsedTime));
}

There may be multiple animations in a gltf model. Can filament support multiple animation combinations at present?
Or is there any other way to do it? thank you very much！

**Comment by prideout** [comment]
If your glTF asset defines animations that affect disjoint sets of nodes, e.g. a "head nod" animation and a "legs walk" animation, then yes those can be applied in combination, simply call `applyAnimation` twice, one for each animation.

If you need to cross-fade between two animations that affect the same set of nodes, e.g. "legs walk" and "legs run", then you can use the new `applyCrossFade` method that we recently added to the `Animator` interface.

**Comment by caorm** [comment]
@prideout 

size_t animationCount = animator->getAnimationCount(); 
for (size_t i = 0; i < animationCount; ++i) { 
animator->applyAnimation(i, static_cast(elapsedTime)); 
}

The code is as above,
I call applyAnimation multiple times like this, once for each animation, but the last animation overwrites the previous animations.

I want to do multiple animations at the same time, we use this URL to combine gltf animations: 

https://gltf-viewer.donmccurdy.com/

the effect is perfect.


use 1.22 filament version.

**Reply by prideout** [reply]
Yes, if those animations affect the same set of nodes, the last one applied will overwrite the previous ones.  If you need to mix two different animations, use applyCrossFade, which was introduced in v1.23.0.
